csrc/host_ir/executor.cpp

```diff
@@ -254,6 +254,10 @@ KernelArgumentHolder HostIrEvaluator::runWithInput(
     const std::unordered_map<Val*, PolymorphicValue>& val_to_PValue) {
   expr_evaluator_ = ExpressionEvaluator();
   expr_evaluator_.bind("numberOfStreams", params_.number_of_streams);
+  FusionGuard fg(container_.get());
+  expr_evaluator_.bind(
+      NamedScalar::getParallelIndex(ParallelType::DIDx),
+      communicator_->deviceId());
```

Comment:
It looks like we are missing deviceIdx.x in this method: https://github.com/NVIDIA/Fuser/blob/d2f333f1284f00e302ff97563e02765257d80893/csrc/ir/nodes.cpp#L4244

Otherwise we can
```suggestion
  expr_evaluator_.bind(
      "deviceIdx.x",
      communicator_->deviceId());
```

Is that correct?

Reply:
outdated https://github.com/NVIDIA/Fuser/pull/4515#discussion_r2121345543